js/src/wp-seo-replacevar-plugin.js

```diff
@@ -127,6 +129,59 @@ import "./helpers/babel-polyfill";
 		}
 	};
 
+	/**
+	 * Subscribes to Gutenberg to watch a possible parent page change.
+	 *
+	 * @returns {void}
+	 */
+	YoastReplaceVarPlugin.prototype.subscribeToGutenberg = function() {
+		if ( ! isGutenbergDataAvailable() ) {
+			return;
+		}
+
+		let fetchedParents = { 0: "" };
+		let currentParent  = null;
+		const wpData       = window.wp.data;
+		wpData.subscribe( () => {
+			let newParent = wpData.select( "core/editor" ).getEditedPostAttribute( "parent" );
+
+			if ( typeof newParent === "undefined" || currentParent === newParent ) {
+				return;
+			}
+
+			currentParent = newParent;
+
+			if ( newParent < 1 ) {
+				this._currentParentPageTitle = "";
+
+				this.declareReloaded();
+
+				return;
+			}
+
+			if ( ! isUndefined( fetchedParents[ newParent ] ) ) {
+				this._currentParentPageTitle = fetchedParents[ newParent ];
+
+				return;
+			}
+
+			const page = new wp.api.models.Page( { id: newParent } );
+			page.fetch().then(
```

Comment:
My only doubt is why we're using `then()` instead of `done()`.
Also, I may be wrong, but to my understanding `fail()` runs only if the user is offline or other network errors occur.

Reply:
The `fail()` handles a bad response, this will be the case when a page doesn't exists ( 404 response status). The use of `then` is because I used the way I knew for using Promises. Do you want me to change it in `done`?

Reply:
I guess `then()` vs `done()` is more a stylistic issue, as to my understanding `then()` with just a "done" callback is equivalent. About `fail()`, if a page doesn't exist it won't show up in the"Parent Page" dropdown, right? Unless I'm missing something. Also, the code is already returning early is newParent doesn't exist (ID less then 1)

Reply:
I agree. An unexisting page is one of the cases. It might be possible that a page got removed during the page edit or an server error occurs. Personally I don't trust the behaviour and values of DOM-elements because it can be changed with JavaScript. That's why I used the `fail`.

Reply:
OK thanks 🙂 then it can be considered a super-safe check, which is good.